src/public_key.rs

```diff
@@ -383,6 +383,19 @@ impl PublicKey {
     }
 }
 
+#[cfg(feature = "solana")]
+impl TryFrom<PublicKey> for solana_sdk::pubkey::Pubkey {
+    type Error = &'static str;
+
+    fn try_from(public_key: PublicKey) -> std::result::Result<Self, Self::Error> {
+        if let PublicKeyRepr::Ed25519(key) = public_key.inner {
+            Ok(solana_sdk::pubkey::Pubkey::new(key.as_ref()))
+        } else {
+            Err("Only Helium Ed25519 keys may be converted to a Solana address")
```

Comment:
Not quite the error I was aiming for.. Can we actually have an `DecodeError::Unsupported` enum entry instead?

Reply:
Fixed